wordpress-mobile/WordPress-Android
# Retrieve is-new-post flag when switching editors

Fixes https://github.com/wordpress-mobile/gutenberg-mobile/issues/2072 reported on the gutenberg-mobile repo.

Reinstates the value of the `mIsNewPost` flag that is captured in the Intent's extras but wasn't picked up.

This PR includes a UI test to verify that the Preview screen is launched but, landing it as commented out to match https://github.com/wordpress-mobile/WordPress-Android/pull/11530. Also note that since the network requests/responses are mocked, the preview itself is not valid... the test only tests if the screen is launched, verifying that the original issue is fixed.

✅I'm opening this PR as Draft as I'm not sure yet if the value setting will disrupt other call sites that depend on `mIsNewPost`'s value.

I checked so far:
1. `mEditPostRepository` has a valid post, is set to "DRAFT" state, and has a snapshot saved, [just like the lines in `newPostSetup()`](https://github.com/wordpress-mobile/WordPress-Android/blob/8f0e676e8556ed8d51852a53a416d4584cfc9c3a/WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java#L358-L364).

### To test A
1. Start a new post in the block editor
2. Put in a title
3. Hit the 3-dots menu and "Switch to classic editor"
4. Hit the 3-dots menu again and "Preview"
5. After a bit of time, the post preview should be visible in an in-app browser

### To test B
1. Uncomment everything in https://github.com/wordpress-mobile/WordPress-Android/blob/8e4a56cea0cf3014d13c65e1fad2eeb403ddcb84/WordPress/src/androidTest/java/org/wordpress/android/e2e/BlockEditorTests.java to enable the block editor UI test
2. Issue `adb shell am instrument -w -e class 'org.wordpress.android.e2e.BlockEditorTests#testSwitchToClassicAndPreview' org.wordpress.android.beta.test/org.wordpress.android.WordPressTestRunner` on the CLI to run the specific test. It should succeed.

PR submission checklist:
